tests/t5_cp.py

```diff
@@ -59,3 +59,62 @@ def tst_cp(self):
 
         finally:
             shutil.rmtree(tempdir)
+
+
+    def test_cp_inode_invalidate(self):
+        if os.getuid() != 0:
+            pytest.skip('test_cp_inode_invalidate requires root, skipping.')
+
+        self.passphrase = None
+        self.mkfs()
+
+        # Create monkeypatched mount.s3ql
+        # Override pyfuse3.invalidate_inode : Drop kernel dentries and inodes cache
+        # just before calling pyfuse3.invalidate_inode
+        patched_mount = """
+import sys
+import os.path
+
+basedir = os.path.abspath(os.path.join(os.path.dirname(sys.argv[0]), '..'))
+sys.path = [os.path.join(basedir, 'src')] + sys.path
+
+import pyfuse3
+from _pytest.monkeypatch import MonkeyPatch
+pyfuse3_invalidate_inode = pyfuse3.invalidate_inode
+def patched_pyfuse3_invalidate_inode(inode):
+    # echo 2 > /proc/sys/vm/drop_caches : Drop kernel dentries and inodes cache
+    drop_caches = open("/proc/sys/vm/drop_caches", "w")
```

Comment:
Plese use a `with` statement to ensure this gets closed again.

Reply:
:+1: